lib/expand/join.go

```diff
@@ -1022,6 +1027,10 @@ func (p *Peer) createShrinkOperation(ctx context.Context) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
+	if len(operation.Servers) == 0 {
+		p.WithField("operation", operation.String()).Warn("Failed to create shrink for invalid operation.")
```

Comment:
Nit: I think we might never see this log message but still, instead of saying "for invalid operation" I'd say explicitly that servers are missing from the operation state.

Reply:
Oh, you can see the message if you abort the expand operation that gets stuck half-way - i.e. if the controller cannot reach the joining node to properly accept the agent.